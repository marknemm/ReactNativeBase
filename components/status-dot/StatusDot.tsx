import { View } from 'react-native';
import { useStyles } from './StatusDot.styles';
import type { StatusDotProps } from './StatusDot.interfaces';

/**
 * A colored dot to represent a status.
 *
 * @param props The component {@link StatusDotProps}.
 * @returns The {@link StatusDot} component.
 */
const StatusDot: React.FC<StatusDotProps> = ({
  color,
  size,
  style,
  testID = 'rnb-status-dot',
}) => {
  const styles = useStyles({ color, size, style });

  return (
    <View
      style={styles.statusDot}
      testID={testID}
    />
  );
};

export type * from './StatusDot.interfaces';
export default StatusDot;
