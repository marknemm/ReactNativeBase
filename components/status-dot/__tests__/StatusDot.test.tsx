import StatusDot from '@components/status-dot/StatusDot';
import AppProvider from '@test/contexts/app/AppProvider';
import { render, screen } from '@testing-library/react-native';

describe('<StatusDot />', () => {
  describe('styles', () => {
    it('renders with default styles', () => {
      render(
        <StatusDot />,
        { wrapper: AppProvider }
      );

      const statusDotView = screen.getByTestId('rnb-status-dot');

      expect(statusDotView).toHaveStyle({
        width: 10,
        height: 10,
        borderRadius: 5,
        backgroundColor: 'gray',
      });
    });

    it('renders with custom styles', () => {
      render(
        <StatusDot
          color="green"
          size={20}
          style={{ marginLeft: 10 }}
        />,
        { wrapper: AppProvider }
      );

      const statusDotView = screen.getByTestId('rnb-status-dot');

      expect(statusDotView).toHaveStyle({
        width: 20,
        height: 20,
        borderRadius: 10,
        backgroundColor: 'green',
        marginLeft: 10,
      });
    });
  });

  describe('snapshots', () => {
    it('renders correctly', () => {
      render(
        <StatusDot />,
        { wrapper: AppProvider }
      );

      expect(screen.toJSON()).toMatchSnapshot();
    });
  });
});
