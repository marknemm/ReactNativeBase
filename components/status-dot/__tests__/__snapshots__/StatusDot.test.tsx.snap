// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StatusDot /> snapshots renders correctly 1`] = `
<View
  style={
    {
      "backgroundColor": "gray",
      "borderRadius": 5,
      "height": 10,
      "width": 10,
    }
  }
  testID="rnb-status-dot"
/>
`;
